File: 1000_DEV/ttlive-web/src/rest/data/ChatMessage.ts
export interface ChatMessage {
    id: number;
    username: string;
    text: string;
    createdAt: string;
}

export interface RequestChatMessage {
    username: string;
    text: string;
}

export function sortChatMessages(messages: Array<ChatMessage>) {
    return messages.sort((a, b) => a.createdAt === b.createdAt ? 0 : a.createdAt > b.createdAt ? 1 : -1);
}

File: 1000_DEV/ttlive-ejb/src/main/java/com/ttlive/bo/ChatMessage.java
package com.ttlive.bo;

import java.time.LocalDateTime;

import com.ttlive.persistence.entity.ChatMessageEntity;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class ChatMessage {
	private long id;	
	private String text;
	private boolean isEditor;
	private String username;
	private LocalDateTime createdAt;
	
	public static class ChatMessageBuilder {
		public ChatMessageBuilder entity(ChatMessageEntity entity) {
			this.id = entity.getId();
			this.text = entity.getText();
			this.isEditor = entity.isEditor();
			this.username = entity.getUsername();
			this.createdAt = entity.getCreatedAt();
			return this;
		}
	}
}
